booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,645 @@
+      BookingRole::isShipper,
+      matched,
+      HttpMessageType.REQUEST,
+      checks
+    );
+  }
+  private static final JsonContentCheck CHECK_EXPECTED_DEPARTURE_DATE = JsonAttribute.customValidator(
+    "Check expected departure date can not be past date",
+    (body) -> {
+      String expectedDepartureDate = body.path("expectedDepartureDate").asText("");
+      var invalidDates = new LinkedHashSet<String>();
+      if(!expectedDepartureDate.isEmpty()) {
+        LocalDate expectedDepartureLocalDate = LocalDate.parse(expectedDepartureDate);
+        if (expectedDepartureLocalDate.isBefore(LocalDate.now())) {
+          invalidDates.add(expectedDepartureLocalDate.toString());
+        }
+      }
+      return invalidDates.stream()
+        .map("The expected departure date '%s' can not be past date"::formatted)
+        .collect(Collectors.toSet());
+    }
+  );
+
+  private static final JsonContentCheck CHECK_EXPECTED_ARRIVAL_POD = JsonAttribute.customValidator(
+    "Check expected arrival dates are valid",
+    (body) -> {
+      String providedArrivalStartDate = body.path("expectedArrivalAtPlaceOfDeliveryStartDate").asText("");
+      String providedArrivalEndDate = body.path("expectedArrivalAtPlaceOfDeliveryEndDate").asText("");
+      var invalidDates = new LinkedHashSet<String>();
+      if (!providedArrivalStartDate.isEmpty() && !providedArrivalEndDate.isEmpty()) {
+        LocalDate arrivalStartDate = LocalDate.parse(providedArrivalStartDate);
+        LocalDate arrivalEndDate = LocalDate.parse(providedArrivalEndDate);
+        if (arrivalStartDate.isAfter(arrivalEndDate)) {
+          invalidDates.add(arrivalStartDate.toString());
+        }
+        if (arrivalEndDate.isBefore(arrivalStartDate)) {
+          invalidDates.add(arrivalEndDate.toString());
+        }
+      }
+      return invalidDates.stream()
+        .map("The expected arrival dates  '%s' are valid"::formatted)
+        .collect(Collectors.toSet());
+    }
+  );
+
+  private static final Predicate<JsonNode> IS_ISO_EQUIPMENT_CONTAINER_REEFER = (uteNode) -> {
+    var isoEquipmentNode = uteNode.path("ISOEquipmentCode");
+    return isReeferContainerSizeTypeCode(isoEquipmentNode.asText(""));
+  };
+  private static final Predicate<JsonNode> HAS_ISO_EQUIPMENT_CODE = (reqEquipNode) -> {
+    var isoEquipmentNode = reqEquipNode.path("ISOEquipmentCode");
+    return isoEquipmentNode.isTextual();
+  };
+  private static final Predicate<JsonNode> IS_ACTIVE_REEFER_SETTINGS_REQUIRED = (reqEquipNode) -> {
+    var norNode = reqEquipNode.path("isNonOperatingReefer");
+    if (norNode.isMissingNode() || !norNode.isBoolean()) {
+      // Only require the reefer if there is no equipment code or the equipment code is clearly a reefer.
+      // Otherwise, we give conflicting results in some scenarios.
+      return !HAS_ISO_EQUIPMENT_CODE.test(reqEquipNode) || IS_ISO_EQUIPMENT_CONTAINER_REEFER.test(reqEquipNode);
+    }
+    return !norNode.asBoolean(false);
+  };
```

Comment:
I am not sure this is correct.
It is correct that an `Active Reefer Settings` is required if `isNonOperatingReefer` is set to `false`. If `isNonOperatingReefer` is missing or if it is `true` then there cannot be a `Active Reefer Setting`. To me it looks like you allow a `Active Reefer Setting` if `isNonOperatingReefer` is missing